apps/launch/hooks/useGetMyStakingContracts.ts

```diff
@@ -0,0 +1,175 @@
+
+  try {
+    const uri = `${HASH_PREFIX}${tokenUri.substring(2)}`;
+    const ipfsUrl = `${GATEWAY_URL}${uri}`;
+    const response = await fetch(ipfsUrl);
+    const json = await response.json();
+    return json;
+  } catch (e) {
+    window.console.error(e);
+  }
+
+  return null;
+};
+
+const useGetAllNominees = () => {
+  // it is only available on mainnet, hence hardcoded.
+  const { data: nominees } = useReadContract({
+    address: (VOTE_WEIGHTING.addresses as Record<number, Address>)[mainnet.id],
+    abi: VOTE_WEIGHTING.abi,
+    chainId: mainnet.id,
+    functionName: 'getAllNominees',
+    args: [],
+    query: {
+      select: (data: unknown) => {
+        const list = data as { account: Address; chainId: bigint }[];
+        return list.filter((nominee) => nominee.chainId === BigInt(mainnet.id));
+      },
+    },
+  });
+
+  return nominees;
+};
+
+const useGetMyStakingContractsMetadata = (addresses: Address[]) => {
+  const { chainId } = useAccount();
+  const [metadata, setMetadata] = useState<Metadata[]>([]);
+
+  const contracts = addresses.map((address) => ({
+    address,
+    abi: STAKING_TOKEN.abi as Abi,
+    chainId,
+    functionName: 'metadataHash',
+    args: [],
+  }));
+
+  const { data } = useReadContracts({
+    contracts,
+    query: {
+      enabled: addresses.length > 0,
+      select: async (list) => {
+        const metadataHashList = list.map(({ result }) => result as string);
+        try {
+          const metadataList = await Promise.allSettled<Metadata>(
+            metadataHashList.map((hash) => getMetadata(hash)),
+          );
+
+          const results: Metadata[] = [];
+          for (const response of metadataList) {
+            if (response.status === 'fulfilled' && response.value) {
+              results.push(response.value);
+            }
+          }
+
+          return results.filter((metadata) => !!metadata);
+        } catch (error) {
+          window.console.error(error);
+        }
+
+        return [];
+      },
+    },
+  });
+
+  useEffect(() => {
+    (async () => {
+      if (!data) return;
+
+      try {
+        const result = await data;
+        setMetadata(result);
+      } catch (error) {
+        window.console.error(error);
+      }
+    })();
+  }, [data, addresses]);
+
+  return metadata;
+};
+
+const useGetInstanceAddresses = () => {
+  const client = usePublicClient();
+  const { chainId } = useAccount();
+
+  const [instanceAddresses, setInstanceAddresses] = useState<Address[]>([]);
+
+  const currentChainId = chainId as SupportedChain;
+
+  useEffect(() => {
+    (async () => {
+      if (!currentChainId) return;
+      if (!client) return;
+
+      try {
+        const eventLogs = await client.getLogs({
+          address: STAKING_FACTORY.addresses[`${currentChainId}`] as Address,
+          event: parseAbiItem(
+            'event InstanceCreated(address indexed, address indexed, address indexed)',
+          ),
+          fromBlock: BigInt(0),
```

Comment:
can we hardcode the block to the contract deployed one? can change when real contracts will be deployed.

didn't find the exact blocks, but the closes are:
Polygon: 57751430,
Gnosis: 34225443
Ethereum: 20009952

Reply:
1) in my virtual test nets
2) I deployed them myself, for testing. They are not deployed yet in real blockchain (neither factories, nor staking tokens, vote weighting, etc.)